packages/block-library/src/navigation/edit.js

```diff
@@ -70,7 +114,15 @@ function Navigation( {
 			[ `items-justified-${ attributes.itemsJustification }` ]: attributes.itemsJustification,
 			'is-vertical': attributes.orientation === 'vertical',
 			'is-responsive': attributes.isResponsive,
+			'has-text-color': !! textColor.color || !! textColor?.class,
+			[ `has-${ textColor?.slug }-color` ]: !! textColor?.slug,
```

Comment:
Could you use the `getColorClassName` utility here to create the class from the slug? Otherwise, this code would need to use lodash's `kebabCase` to process the color slug. Same for the background color class.

We're tied to lodash's kebabCase implementation because that utility has been in use since the very beginning to create classes from slugs. [This](https://github.com/WordPress/gutenberg/issues/32347) describes a number of edge cases we need to cover. We had to replicate the [same behavior](https://github.com/WordPress/gutenberg/pull/32766) in the server.

A use case where this would break would be:

- theme provides one color with slug `5red`
- in any block that uses the hooks system or the `getColorClassName` utility the class will be created as `has-5-red-color`
- in navigation will be `has-5red-color`

Reply:
Thanks, I didn't know about that.